packages/react-scripts/scripts/utils/verifyTypeScriptSetup.js

```diff
@@ -262,6 +263,14 @@ function verifyTypeScriptSetup() {
     );
   }
 
+  if (parsedTsConfig.paths != null) {
+    if (semver.lt(ts.version, '4.1.0') && parsedTsConfig.baseUrl == null) {
+      messages.push(
+        `${chalk.cyan('paths')} requires ${chalk.cyan('baseUrl')} to be set`
+      );
+    }
+  }
```

Comment:
This is to allow for the fact that TypeScript 4.1 does not need `baseUrl` to use `paths`.